### PR TITLE
Update to latest API/tools, materialize & use appcompat

### PR DESCRIPTION
**Pull request by fat-tire**
(also one instance of each activity + other minor changes)

Might be able to change the minSDK to even lower, whatever appcompat supports.

Enjoy and let me know if something broke for ya.

Edit:  I should add I did this in less than an hour, so it's very possible :)

### CODE REVIEW

**Comment by ocdtrekkie** [comment]
Ewwwwwwwwww. Big -1.

**Comment by fat-tire** [comment]
Wow.  That's constructive.  FWIW it looks 99% the same.

So like, wat?

**Comment by ocdtrekkie** [comment]
AppCompat + Material = A pretty excellent way to wreck a quality Android app.

**Comment by fat-tire** [comment]
Like what are you even talking about?

**Comment by aaronjwood** [comment]
Thanks for submitting this, always love getting feedback and contributions. There are a few things that I think would need to get fixed before I'd merge this in.

screenshot_2015-08-23-22-06-02
The items on the list are now the same size so you don't get the separation of the DNS name being larger than the IP and MAC address. I'd really rather not have everything be the same size of that list.

screenshot_2015-08-23-21-58-25
The colors have changed but I actually don't mind this look. It just kinda breaks the icy blue theme that was everywhere else. It's sorta like having two different styles in there now.

screenshot_2015-08-23-21-58-44
The port range dialog doesn't have any spacing/padding anymore. The colors (especially the button) also don't really match up with the activities and other elements of the application. It's sorta like having two different styles in there now.

Also, how come you are using the `android:launchMode="singleTask"`? Google's docs recommend against that unless there's a special case. I'm not seeing what it provides here.

Everything else looks good. I should have used `sp` like you did for fonts in the first place. And it's definitely nice to be on the latest API/tools.

**Comment by ocdtrekkie** [comment]
No longer Holo, @fat-tire. The weird grey/green stuff in the screenshots above, the stupid ripple animations. Exactly what forking repos is for. :/

And AppCompat is a intentionally just-barely-incompatible way that makes a huge mess of a codebase. It should be avoided unless it's fundamentally needed. I don't see anything in this that justifies changing from building on Android to building on AppCompat.

**Comment by fat-tire** [comment]
Hey Aaron-- first off I think I forgot to mention how frickin' cool this app is-- the speed is ridiculous.

Let me go back manana probably in the evening and look at the few tweaks you mentioned.  I'll add them in a separate commit on this same branch, so they should appear in this pull commit.  Didn't notice the difference in the port range layout...  Well I did as I was changing it when I pulled what lint told me was a useless RelativeLayout, but I thought I'd put it back... anyway I'll take another look and see if I can fix it.  (The colors are just the ones that were in `colors.xml` so i reused them...)

The singletask launchmode I added only after when I was testing and hit the I realized I was unexpectedly creating a zillion activities for the same host/network...  was that how you wanted it?  If so, I can remove it-- I was just thinking it must be eating up some ram/time creating a new activity every time, this pretty much keeps the whole app to two activities...

@ocdtrekkie   "*I don't see anything in this that justifies changing from building on Android to building on AppCompat?*"  "Building on AppCompat"?  Umm, let's save us both some time here-- are you an insane person?  If not, seriously, what the f are you talking about?

**Comment by aaronjwood** [comment]
Glad you like it! I'm planning on adding new features to it soon, I've just been too busy at work.

I don't think you'll need the `singleTask` stuff. There are lots of threads being created for the scan since it's very I/O bound, not activities. I think the application would crash if there were as many activities quickly spawned and killed as there are threads :)

How come you want to extend `AppCompatActivity` instead of `Activity`? What is the main benefit/goal of doing that? The share of devices using versions of Android less than 4.0 are so small nowadays https://developer.android.com/about/dashboards/index.html

**Comment by ocdtrekkie** [comment]
@aaronjwood Once you use Theme.AppCompat, AFAIK, it makes you change to AppCompatActivity and such. That's my whole issue with AppCompat.

**Comment by fat-tire** [comment]
Okay I'll get rid of the `SingleTask`.  I switched to `AppCompatActivity` for a couple of reasons-- one is that the 5% of pre-4.0 Android devices still in use (not targeted yet, but could be) may seem like a small amount proportionally, but with about 1 billion Android users out there, its not insignificant.  (I actually still have two Gingerbread devices I use, admittedly for testing)...

Also, it's not just compatibility features extending straight to v4/v7, you may get some 5.0 features in 4.1, or 6.0 features in 4.4, etc. So as you continue to develop the app, I think you'll find you'll get a lot of compatibility "for free"...  You also get the advantage of being able to use a single unified parent theme for all devices, and bring material and design features along with them to pre-lollipop devices.  Plus, you can simplify lots of xml into a single file across versions (you don't have to track multiple -v14, v-21, etc.), and they're all have the same features/layouts/appearances.  Feature parity even within the 4.x versions can eliminate requiring API checks and branching code if you use the support classes... you can use one, consistent flow that will just work on all the various versions.

Finally, given how great the compat/support stuff has become in the past two years or so, there's really no reason *not* to use it, especially as more and more new features appear moving forward.

Of course, I don't need to mention this, but @ocdtrekkie is talking out of his ass.

**Comment by ocdtrekkie** [comment]
From my understanding, switching to AppCompat bloats the app by essentially takong chunks of Android and shoving them into your app package? For what benefit, @fat-tire? Material, which makes the UI inconsistent with the app's design.

**Comment by fat-tire** [comment]
Okay @aaronjwood , take a look at these changes.  I think I addressed everything you mentioned, so hopefully they work for you.  You now have a theme you can play with specifically for dialogs too.  I tried to match the rest of the app, but If you want to tweak or mess with it or not use it or whatever, feel free.

@ocdtrekkie I don't know where you got the idea that AppCompat is just about material design. Go [read up](https://developer.android.com/tools/support-library/features.html) on what it actually does.  It's about extending *actual functionality*, not just styling, though that's pretty great too.  It means you can add UI like drawers, FABs, accessibility features, you get all the toolbar/action bar stuff, you can use recyclerview and cardview, switches instead of checkboxes, and way more going back to devices that otherwise wouldn't support any of it.  It's a lot of stuff that makes the apps more uniform across devices, but also makes development easier too.  Check out the [revisions](https://developer.android.com/tools/support-library/index.html#revisions) because they list tons of stuff that got back ported and you can use confidently once in your code without having to find workarounds for all the older versions.

Yes, AppCompat/Support *does* supplement the framework in the app, but only where necessary to provide consistent (and often faster/more polished) behavior across versions of Android-  you can use proguard to optimize and strip out much of the unused code, and as I mentioned earlier, space is also saved because you don't have to provide multiple versions of xml files, and you don't need the code to branch all over the place either, and you don't have take the time to re-implement the same type of behavior figuring out at what point some feature you want to use was added and then coding around it...

**Comment by aaronjwood** [comment]
Thanks, I will take a look at this as soon as I can. Might be a few days as work is really busy for me lately :)

File: app/build.gradle [hunk 1]
```diff
@@ -1,8 +1,21 @@
 apply plugin: 'com.android.application'
 
 android {
-    compileSdkVersion 22
-    buildToolsVersion '22.0.1'
+    compileSdkVersion 23
+    buildToolsVersion '23.0.0'
+    useLibrary 'org.apache.http.legacy'
```
Comment (aaronjwood) [inline app/build.gradle]: Is this from the support library? Do we need to use this legacy library?

**Comment by fat-tire** [comment]
If you're talking about the apache httpclient stuff, yes-- it was deprecated in api 22.  See [here](https://stackoverflow.com/questions/30979929/why-were-the-entire-http-apis-in-android-deprecated-in-api-level-22). Google [recommends](https://www.youtube.com/watch?v=yhv8l9F44qo) using [Volley](https://developer.android.com/training/volley/index.html) instead, though there are other options...

That line is necessary to keep the deprecated stuff and avoid breakage..

**Comment by aaronjwood** [comment]
Good to know, thanks. I'll update the code at some point to get rid of the apache http lib

**Comment by fat-tire** [comment]
Cool.  Well I'll leave the rest to you then!  Enjoy, and again, great app!

**Pull request merged**